pkg/controller/stacks/stack/stack.go

```diff
@@ -136,6 +156,14 @@ func (h *stackHandler) sync(ctx context.Context) (reconcile.Result, error) {
 func (h *stackHandler) create(ctx context.Context) (reconcile.Result, error) {
 	h.ext.Status.SetConditions(runtimev1alpha1.Creating())
 
+	// Add the finalizer before the RBAC and Deployments. If the Deployment
+	// irreconcilably fails, the finalizer must be in place to delete the Roles
+	patchCopy := h.ext.DeepCopy()
```

Comment:
By watching the logs and running into errors, I found that the `Update` was failing because the `Stack` had already been changed (by `Status.SetConditions`?).  By using `Patch` this problem was avoided.